zip-0320.rst

```diff
@@ -5,7 +5,347 @@
+understand and send to these addresses.
+
+In addition, by integrating with the Unified Address framework, it becomes
+possible for the addresses being generated to include extra metadata, in
+particular, metadata items such as an Address Expiry Height or Address Expiry
+Date [#zip-0316-address-expiry]_ may be included. For exchange use cases such
+as Binance's, it is useful to ensure that an address provided to a user has a
+limited utility life, such that after expiration the user must obtain a new
+address in order to be able to continue to send funds
+[#binance-address-expiry]_.
+
+Specification (Alternative 2)
+-----------------------------
+
+Upon activation of this ZIP, the section `Encoding of Unified Addresses` of ZIP
+316 [#zip-0316-unified-addresses]_ will be modified to define a new
+Traceable Receiver Type having typecode :math:`\mathtt{0x04}`, the value of
+which MUST be the 20-byte **validating key hash** of a Zcash P2PKH Address as
+defined in [#protocol-transparentaddrencoding]_.
+
+The "Requirements for both Unified Addresses and Unified Viewing Keys" section
+of ZIP 316 [#zip-0316-unified-requirements]_ will be modified as follows —
+the text::
+
+  A Unified Address or Unified Viewing Key MUST contain at least one
+  shielded Item (Typecodes :math:`\mathtt{0x02}` and :math:`\mathtt{0x03}`).
+  The rationale is that the existing P2SH and P2PKH transparent-only
+  address formats, and the existing P2PKH extended public key format,
+  suffice for representing transparent Items and are already supported
+  by the existing ecosystem.
+
+will be replaced by::
+
+  A Unified Address MUST contain at least one Receiver and any number
+  of Metadata Items. The selection of Receivers is further restricted 
+  such that a Unified Address MUST **either** contain at least one shielded
+  Receiver (Typecodes :math:`\mathtt{0x02}` and :math:`\mathtt{0x03}`), OR 
+  MUST contain **only** a Traceable Address Receiver (Typecode
+  :math:`\mathtt{0x04}`).
+
+  A Unified Viewing Key MUST contain at least one shielded Item (Typecodes
+  :math:`\mathtt{0x02}` and :math:`\mathtt{0x03}`). 
+
+A Traceable Unified Address is produced from a mainnet Zcash P2PKH address by
+executing the following steps:
+
+1. Decode the address to a byte array using the Base58Check decoding algorithm
+   [#Base58Check]_.
+2. If the length of the resulting byte sequence is not 22 bytes or if its
+   two-byte address prefix is not :math:`\mathtt{[0x1C, 0xB8]}`, return an
+   error. Otherwise, let the **validating key hash** be the remaining 20 bytes
+   of the array after removing the two-byte address prefix.
+3. Construct a new Unified Address using a single Traceable Receiver
+   :math:`\mathtt{0x04}` with the 20-byte **validating_key_hash** as its value.
+   In addition, metadata items such as an Address Expiry Height or Address
+   Expiry Date [#zip-0316-address-expiry]_ MAY be included.
+
+Wallets and other Senders sending to a Traceable Unified Address MUST ensure
```

Comment:
```suggestion
Wallets and other Senders sending to a Traceable P2PKH Receiver MUST ensure
```
As currently written, a Traceable Unified Address can contain only a Traceable P2PKH Receiver (and metadata). The change here enforces the intended restriction regardless of that design decision, in case it is relaxed in future. Also, it's more accurate to say that funds are sent to a Receiver, not a UA.

Reply:
I think that the concept of a traceable P2PKH receiver is not well-defined, whereas a Traceable Address could in the future be specified to alternatively support a P2SH receiver.

Reply:
That being said, we could define Traceable P2PKH receiver more clearly.

Reply:
It's defined here: https://github.com/zcash/zips/pull/760/files#r1451776286